test/ui-test/testSuites/suite_messaging/tst_ChatFlow/bdd_hooks.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# This file contains hook functions to run as the .feature file is executed
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "../../../testSuites/global_shared/"))
+sys.path.append(os.path.join(os.path.dirname(__file__), "../../../src/"))
+
+from steps.startupSteps import StartupSteps
+
+# Global properties for the specific feature
+_user = "tester123"
+_password = "TesTEr16843/!@00"
+_chat_room = "test"
+
+@OnFeatureStart
+def hook(context):
+    StartupSteps.context_init(context)   
+    StartupSteps.given_a_first_time_user_lands_on_and_generates_new_key(context)
+    StartupSteps.when_the_user_sings_up(_user, _password)
+    StartupSteps.when_the_user_lands_on_the_signed_in_app()
```

Comment:
Could we have one step:
```
when_the_sign_up(_user, _password)
```
Do we need to decouple that way?

Reply:
I've tried to follow the same bdd steps we aready have in our `XXsteps.py` files bc we can need them in some point separately, like for example, in the `singUpSteps.py`, where I've replaced the direct calls to screens to this "intermediate" static methods. I've done this way to try not to write the same code in different ways and avoiding future changes in one side and not in the other.. 

In general, this intermediary static methods should be very specifics, so this new file should not grow a lot, since they should only define steps to get the application into an idle state.

So.. we could do what you propose but then we will have the existing steps as they were (calling directly to screens actions) and the static methods working separately and only with the propose of being used inside the `hooks` for the specific propose of starting up or driving the app into an idle state, and not mixing it in the exisiting `XXsteps.py` definitions.

For me both ways make sense so.. WDYT?

Reply:
There will likely be enough scope within the tests to have both - keep the existing steps so the functions are available when needed to go to a specific stage of the start up and also create a wrapper step which runs the full sequence.

Reply:
The reason for me asking this is that I notice some confusion when getting started with our test, expectation is to have simple login/sign up stage.

Also as a user I would never want to go halfway through the login process this is why I believe they are 1 steps,
ofc behind the scene we can decouple into function.

We can keep it that way for now and re-evaluate later

Reply:
Thanks for sharing your thoughts!

I've finally created a new wrapper functions `signs_up_process_steps` that encapsulates the sequence and it is the one called from the `hooks` file.